Azure/sonic-buildimage
# python http-server host service implementation

Signed-off-by: anamehra <anamehra@cisco.com>

```
# Python HTTP Server service.
# This service run based the configuration provided in
# device/<platform>/platform_env.conf
#
# configuration parameters provided by platform_env.conf
#
# start_http_server: if "yes" or "1", http-server will be started on this
#     node
# http_server_ip   : IP address for http-server. The service also writes this
#     entry in /etc/hosts to map to http-server host name
# http_server_port : Port to bind to, default: 8000
# http_server_dir  : HTTP server home directory path, default: /var/www/
#
###########################################################################
# Copyright (c) 2021 Cisco Systems, Inc. and its affiliates.
# All rights reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#    http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.
#
###########################################################################

# Exampl of config on Supervisor node:
'''
start_http_server=yes
http_server_ip=127.0.0.10
http_server_port=8001
http_server_dir=/var/www/tftp
'''
# Exampl of config on Line Card:
'''
http_server_ip=127.0.0.10
'''
```


#### Why I did it
Enable http-server as host service on Chassis supervisor to enable LCs download image and config files from Supervisor.

#### How I did it
Added http-server as host service.
The service starts a python based http-server on midplane network.

#### How to verify it
Add   device/<platform>/platform_env.conf with following data:
start_http_server=yes
http_server_ip=127.0.0.10
http_server_port=8001
http_server_dir=/var/www/tftp

Boot with the image with above data.

put an image file at  /var/www/tftp/filename
check 'systemctl status http-service'
curl http://http-server:8000/filename -o out_file
compare file_out with /var/www/tftp/filename

#### Which release branch to backport (provide reason below if selected)




#### Description for the changelog

python http-server host service implementation

#### Link to config_db schema for YANG module changes


#### A picture of a cute animal (not mandatory but encouraged)
